lwip: menuconfig option for max number of SNTP servers (IDFGH-5616)

vortigont opened the pull request:
LWIP has two define's for setting max number of sntp servers:
 - total number of handled servers
 - max number of sntp's picked via DHCP

by default both values are equal to 1, but could be set separately.
This is a followup for espressif/arduino-esp32#5343. Having this menuconfig option allows building lwip for arduino-esp32 with multiple sntps's supported

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/esp-idf?pullRequest=7336) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/esp-idf?pullRequest=7336) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/esp-idf?pullRequest=7336) it.</sub>

Alvin1Zhang commented:
Thanks for your contribution.

david-cermak commented on components/lwip/port/esp32/include/lwipopts.h:
```diff
@@ -1074,6 +1074,15 @@ LWIP_FORWARD_DECLARE_C_CXX void sntp_sync_time(struct timeval *tv);
 LWIP_FORWARD_DECLARE_C_CXX uint32_t sntp_get_sync_interval(void);
 
+// Max number of SNTP servers handled (default equal to LWIP_DHCP_MAX_NTP_SERVERS)
+#if defined CONFIG_LWIP_SNTP_MAX_SERVERS && !defined SNTP_MAX_SERVERS
```
```suggestion
#if defined CONFIG_LWIP_SNTP_MAX_SERVERS
```
As we don't allow to override `CONFIG_...` macros defined in configuration menu.

vortigont replied:
OK, I will amend the change into last commit to keep it clean.

david-cermak commented on components/lwip/Kconfig:
```diff
@@ -846,6 +846,23 @@ menu "LWIP"
+            depends on LWIP_DHCP_GET_NTP_SRV
+            help
+                Set maximum number of NTP servers aquired via DHCP-offer.
+                Must be less or equal to total "Maximum number of NTP servers".
```
Question: Does the lwip throws a compile `#error` if it `> SNTP_MAX_SERVERS` ? maybe we should if it doesn't.

vortigont replied:
No, it does not. Actually if I get this right any extra DHCP-servers would be just silently ignored. Not sure that throwing an error and stop here is a good idea.
@david-cermak I can make a test setup with dhcp-server and confirm this if you like?

david-cermak reviewed (approved):
LGTM, just a question and an optional suggestion.

vortigont commented:
@david-cermak I've updated notes for menuoption, also added NTP  debug option and sntp example to utilize SNTP via DHCP if build-in.
Would appreciate if you could review changes.
Thanks!

david-cermak commented:
@vortigont Thanks for the prompt update and addressing the concern of misconfiguration.
This PR looks very good to me and has been added to the internal round of testing and reviews (this usually takes some time though till it gets merged to master).
